### PR TITLE
Integrate documentation.js

### PR DESCRIPTION
**Pull request by mxstbr**
This is so nice! Looks like a huge PR, but is actually much less code now since the theme is no longer inlined. I was able to remove almost all JSDoc comments (except for the examples) in favor of flow types, and documentation understood everything perfectly! 😱 So super cool!

This is what it looks like right now, which to be honest I quite like!

screen shot 2016-12-08 at 21 16 19

Whatcha think @bhough?

Ref: #31

### CODE REVIEW

**Comment by bhough** [comment]
@mxstbr this looks great! Looks like it will need #38 to pass CI, with that and conflict resolution let's merge it ASAP.

**Comment by mxstbr** [comment]
Let's 🚢 this first thing @bhough! Can you review & merge?

File: package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "babel-preset-latest": "^6.16.0",
     "cz-conventional-changelog": "^1.2.0",
     "docdash": "^0.4.0",
+    "documentation": "^4.0.0-beta16",
```
Comment (bhough) [inline package.json]: Should we remove the JSDocs and associated Babel plugin as a dependency now?

Comment (mxstbr) [inline package.json]: Arg yes we should, good catch! Mind doing that and pushing to the branch?

**Pull request merged**